packages/ckeditor5-paste-from-office/src/filters/table.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * @license Copyright (c) 2003-2023, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module paste-from-office/filters/parse
+ */
+import type { UpcastWriter, ViewDocumentFragment } from 'ckeditor5/src/engine';
+
+/**
+ * Set alignment for table pasted from MS Word.
+ *
+ * @param documentFragment The view structure to be transformed.
+ */
+export function tableAlignmentFilter( documentFragment: ViewDocumentFragment, writer: UpcastWriter ): void {
+	for ( const item of documentFragment.getChildren() ) {
+		// If table is not wrapped in div[align] it should be alligned left.
+		// More details:(https://github.com/ckeditor/ckeditor5/issues/8752#issuecomment-1623507171).
+		if ( item.is( 'element', 'table' ) ) {
+			writer.setAttribute( 'align', 'left', item );
+			continue;
+		}
+
+		const align = item.is( 'element' ) && item.getAttribute( 'align' );
```

Comment:
You're using `item.is( 'element' )` check extensively and AFAICS this logic indeed only applies to item that is an element.

In that case it make sense to do a check:

```javascript
if ( !item.is( 'element' ) ) {
	continue;
}
```

At the beginning of for loop.

It will also simplify `item.is( 'element', 'table' )` check to => `item.name === 'table'`.